internal/test/framework/framework.go

```diff
@@ -242,6 +243,67 @@ func (f *TestFramework) getClusterVersion() error {
 	return nil
 }
 
+// GetK8sVersion gets the Kubernetes version in vx.x.x format
+func (f *TestFramework) GetK8sVersion() (string, error) {
+	versionInfo, err := f.OSConfigClient.Discovery().ServerVersion()
+	if err != nil {
+		return "", fmt.Errorf("error while retrieving k8s version : %s", err)
+	}
+
+	return versionInfo.GitVersion, nil
+
+}
+
+// Get KubeNodeUrl gets the url for kubernetes windows node binary.
+func (f *TestFramework) GetKubeNodeURL() (string, error) {
+	k8sVersion, err := f.GetK8sVersion()
+
+	if err != nil {
+		return "", err
+	}
+
+	return "https://dl.k8s.io/" + k8sVersion + "/kubernetes-node-windows-amd64.tar.gz", nil
+}
+
+// GetKubeNodeSHA gets the SHA value for kube node binary as specified in the changelog
+func (f *TestFramework) GetKubeNodeSHA() (string, error) {
+	k8sVersion, err := f.GetK8sVersion()
+
+	if err != nil {
+		return "", err
+	}
+
+	k8sVersionWithoutPatch := strings.Join(strings.Split(k8sVersion[1:], ".")[:2], ".")
+
+	// Retrieve changelog content
+	changeLog, err := http.Get("https://github.com/kubernetes/kubernetes/blob/release-" + k8sVersionWithoutPatch +
+		"/CHANGELOG/CHANGELOG-" + k8sVersionWithoutPatch + ".md")
+
+	if err != nil {
+		return "", fmt.Errorf("error while getting changelog file")
+	}
+
+	changeLogContent, _ := ioutil.ReadAll(changeLog.Body)
+	defer changeLog.Body.Close()
+
+	// Extract binary and corresponding SHA value
+	re := regexp.MustCompile("<tr>\n<td><a href=\"https://dl.k8s.io/" + k8sVersion +
```

Comment:
We've started moving away from re? Can't you do a strings.Split() based on the "/" and get to the value you need.